packages/server/src/filter-test.ts

```diff
@@ -0,0 +1,33 @@
+import { Test } from '@bigtest/suite';
+import * as path from 'path';
+import * as fs from 'fs';
+
+interface Options {
+  files: string[];
+}
+
+function comparePaths(a: string, b: string) {
+  return path.resolve(a) === path.resolve(b);
+}
+
+export function filterTest(test: Test, options: Options): Test {
+  let children: Test[] = [];
+
+  if(options.files.length) {
+    for(let file of options.files) {
+      let child = test.children.find((c) => c.path && comparePaths(c.path, file));
+
+      if(child) {
+        children.push(child);
+      } else if(fs.existsSync(file)) {
+        throw new Error(`file with path '${path.resolve(file)}' exists, but is not part of your test files, if you want to run this test, adjust the 'testFiles' setting in 'bigtest.json'`)
+      } else {
+        throw new Error(`file with path '${path.resolve(file)}' does not exist`);
+      }
+    };
+  } else {
+    children = test.children;
+  }
+
+  return { ...test, children };
+}
```

Comment:
I wasn't entirely sure how to write this. Fundamentally all `Test` nodes are the same, and it shouldn't really affect the filtering at which *level* of nesting a test occurs. However, the `path` property only exists at the *second* level of nesting (just below the root). I thought about various ways of trying to detect whether there is a `path` property, but this all felt pretty brittle, so in the end I opted to just hard-code that we're only filtering the second level. Still not super happy with this.

Reply:
It's really cool that since the orchestrator and agent only think in terms of a single tree node, we can implement this feature without changing the agent at all.

Note that with the advent of #448, the root node _will_ actually have a file path. I guess this means that we won't run any assertions from the root test if it isn't in the list?

Anyway, I think we will be "eventually consistent" 😆

Reply:
This is only partially the case. It works for this feature, since we're only filtering by test, so the set of lanes that we run actually changes. However, if we were to filter by step or assertion, then we would need to make some changes to the agent, because we currently run all of the steps/assertions in a lane, and we take this information from the manifest, not from the tree provided as part of the test run command.

Reply:
Ah yes. Well, it feels like the right change to make to the agent to run the lane according to the tree and only use the manifest as an index to look up steps and assertions.